From 32a17bc25228f387f9c3e10a7f6f1e72dd066c63 Mon Sep 17 00:00:00 2001
From: chewitt <christianshewitt@gmail.com>
Date: Tue, 24 Dec 2019 05:50:45 +0000
Subject: [PATCH] HACK: mmc: meson-gx: force max_segs/max_blk_count values in
 dram access quirk

Temporary workaround for the G12A and G12B SDIO chip bug that causes sdio
issues with brcmfmac. Patch is based on [1] by @hyphop (Khadas) and hints
from @superna9999.

[1] https://github.com/hyphop/khadas-linux-kernel/blob/master/patches/linux-5.4.5/0005-VIM3_wifi_hack.patch

Signed-off-by: Christian Hewitt <christianshewitt@gmail.com>
---
 drivers/mmc/host/meson-gx-mmc.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/mmc/host/meson-gx-mmc.c b/drivers/mmc/host/meson-gx-mmc.c
index e712315..fdf7304 100644
--- a/drivers/mmc/host/meson-gx-mmc.c
+++ b/drivers/mmc/host/meson-gx-mmc.c
@@ -1152,8 +1152,10 @@ static int meson_mmc_probe(struct platform_device *pdev)
 	mmc->caps |= MMC_CAP_CMD23;
 	if (host->dram_access_quirk) {
 		/* Limit to the available sram memory */
-		mmc->max_segs = SD_EMMC_SRAM_DATA_BUF_LEN / mmc->max_blk_size;
-		mmc->max_blk_count = mmc->max_segs;
+		//mmc->max_segs = SD_EMMC_SRAM_DATA_BUF_LEN / mmc->max_blk_size;
+		//mmc->max_blk_count = mmc->max_segs;
+		mmc->max_segs = 1;
+		mmc->max_blk_count = 2;
 	} else {
 		mmc->max_blk_count = CMD_CFG_LENGTH_MASK;
 		mmc->max_segs = SD_EMMC_DESC_BUF_LEN /
-- 
2.7.4

